mailpoet/tests/integration/Newsletter/AutomatedLatestContentAPITest.php

```diff
@@ -0,0 +1,228 @@
+  private $wp;
+
+  private static $alcBlock = [
+    'type' => 'automatedLatestContentLayout',
+    'withLayout' => true,
+    'amount' => '2',
+    'contentType' => 'post',
+    'terms' => [],
+    'inclusionType' => 'include',
+    'displayType' => 'excerpt',
+    'titleFormat' => 'h2',
+    'titleAlignment' => 'left',
+    'titleIsLink' => false,
+    'imageFullWidth' => true,
+    'titlePosition' => 'abovePost',
+    'featuredImagePosition' => 'left',
+    'fullPostFeaturedImagePosition' => 'none',
+    'showAuthor' => 'no',
+    'authorPrecededBy' => 'Author:',
+    'showCategories' => 'no',
+    'categoriesPrecededBy' => 'Categories:',
+    'readMoreType' => 'button',
+    'readMoreText' => 'Read more',
+    'readMoreButton' => [
+      'type' => 'button',
+      'text' => 'Read more',
+      'url' => '[postLink]',
+      'styles' => [
+        'block' => [
+          'backgroundColor' => '#e2973f',
+          'borderColor' => '#e2973f',
+          'borderWidth' => '0px',
+          'borderRadius' => '5px',
+          'borderStyle' => 'solid',
+          'width' => '110px',
+          'lineHeight' => '40px',
+          'fontColor' => '#ffffff',
+          'fontFamily' => 'Arial',
+          'fontSize' => '14px',
+          'fontWeight' => 'bold',
+          'textAlign' => 'left',
+        ],
+      ],
+      'context' => 'automatedLatestContentLayout.readMoreButton',
+    ],
+    'sortBy' => 'newest',
+    'showDivider' => false,
+    'divider' => [
+      'type' => 'divider',
+      'styles' => [
+        'block' => [
+          'backgroundColor' => 'transparent',
+          'padding' => '13px',
+          'borderStyle' => 'solid',
+          'borderWidth' => '3px',
+          'borderColor' => '#aaaaaa',
+        ],
+      ],
+      'context' => 'automatedLatestContentLayout.divider',
+    ],
+    'backgroundColor' => '#ffffff',
+    'backgroundColorAlternate' => '#eeeeee',
+  ];
+
+  public function _before() {
+    parent::_before();
+    $this->wp = $this->diContainer->get(WPFunctions::class);
+    $loggerFactory = $this->diContainer->get(LoggerFactory::class);
+
+    $alc = Stub::make($this->diContainer->get(AutomatedLatestContent::class), [
+      "loggerFactory" => $loggerFactory,
+      "wp" => $this->wp,
+      "transformPosts" => function ($block, $post) {
+        return $post;
+      },
+    ]);
+    $apiPermissionHelper = $this->diContainer->get(APIPermissionHelper::class);
+    $this->alcAPI = new ALCAPI($alc, $apiPermissionHelper, $this->wp);
+
+  }
+
+  public function tearDown(): void {
+    parent::tearDown();
+
+    foreach ($this->createdUsers as $user) {
+      wp_delete_user($user->ID);
+    }
+
+    $this->deleteAllPosts();
+  }
+
+  private function loginWithRole(string $role): void {
+    $username = uniqid("testUser");
+    $email = "$username@test.com";
+    $existingUser = $this->wp->getUserBy("email", $email);
+
+    if ($existingUser) {
+      wp_delete_user($existingUser->ID);
+    }
```

Comment:
Maybe this could be in before/after cleanups?

Reply:
No, this is done to make sure this user is not existing before trying to add it, but I'll add user cleanup anyway

Reply:
> No
Well, it can be in `_before` but I guess the username needs to be fixed and not random. It was a rhetorical question as we don't usually do these things in the tests themselves.